lib/Transforms/CMakeLists.txt

```diff
@@ -1,5 +1,6 @@
 add_circt_library(CIRCTTransforms
   FlattenMemRefs.cpp
+  InsertMergeBlocks.cpp
```

Comment:
Is this the appropriate directory for this transform?  It doesn't seem a general transform for all dialects.

Reply:
I wasn't sure where to place it, as it is a special canonicalization of the `ControlFlow` dialect. 
It might be a reasonable idea to upstream this at some point in time, but I'm not sure if there are other use-cases for this. 
Alternatively, we can move this to the `StandardToHandshake` conversion, as it is currently only used there.